binary_search_tree/binary_search_tree.py

```diff
@@ -5,13 +5,59 @@ def __init__(self, value):
     self.right = None
 
   def insert(self, value):
-    pass
+    # check if the new node's value is less than our current node's value
+    if value < self.value:
+      # if there's no left child here already, place the new node here
+      if not self.left:
+        self.left = BinarySearchTree(value)
+      else:
+        # otherwise, repeat the process!
+        self.left.insert(value)
+    # check if the new node's value is greater than or equal to our current node's value
+    elif value >= self.value:
+      # if there's no right child here already, place the new node here
+      if not self.right:
+        self.right = BinarySearchTree(value)
+      else:
+        # otherwise, repeat the process!
+        self.right.insert(value)
 
   def contains(self, target):
-    pass
+    if target == self.value:
+       return True
+
+    if target < self.value :
+      if not self.left:
+        pass # do nothing because it is recursive loop
+      else:
+        if self.left.contains(target):
+          return True
+
+    elif target > self.value:
+      if not self.right:
+        pass
+      else:
+        if self.right.contains(target) :
+          return True
+
 
   def get_max(self):
-    pass
+    if self.right :
+      # return right most max value
+      return self.right.get_max()
+    else:
+      # return root value
+      return self.value
+    
 
   def for_each(self, cb):
-    pass
\ No newline at end of file
+
+    if self.left:
+      print(f'left {self.left.value}')
+      self.left.for_each(self.value)
+    
+    if self.right:
+      print(f'right {self.right.value}')
+      self.right.for_each(self.value)
+
```

Comment:
doesnt look like you are using the callback function here. are the tests passing?